### PR TITLE
maintain shapes for `ang2vec` and `vec2ang` functions for batched inputs

### PR DESCRIPTION
**Pull request by ASKabalan**
This pull request improves support for multi-dimensional (batch) inputs in the core HEALPix coordinate and neighbor functions, ensuring that all relevant functions preserve batch shapes and work seamlessly with arrays of arbitrary shape. The changes also update the documentation and add comprehensive tests to verify correct behavior for batched inputs.

**Batch shape support and documentation updates:**

* Updated `ang2vec`, `pix2vec`, and `vec2ang` to return outputs with shapes that preserve input batch dimensions, using `jnp.stack(..., axis=-1)` instead of `.T` and updating docstrings to clarify output shapes. [[1]](diffhunk://#diff-b3225420bd8ccd0828bfb2092fa0da2adf6a1a2692f33aebd950eb866a1494f4L1465-R1465) [[2]](diffhunk://#diff-b3225420bd8ccd0828bfb2092fa0da2adf6a1a2692f33aebd950eb866a1494f4L1485-R1485) [[3]](diffhunk://#diff-b3225420bd8ccd0828bfb2092fa0da2adf6a1a2692f33aebd950eb866a1494f4L1500-R1499) [[4]](diffhunk://#diff-b3225420bd8ccd0828bfb2092fa0da2adf6a1a2692f33aebd950eb866a1494f4L1510-R1526)
* Refactored `get_all_neighbours` and internal neighbor-finding logic to operate on arrays with arbitrary batch dimensions, ensuring outputs have shapes like `(8, *batch)` or `(9, *batch)` as appropriate. [[1]](diffhunk://#diff-b3225420bd8ccd0828bfb2092fa0da2adf6a1a2692f33aebd950eb866a1494f4L2179-R2199) [[2]](diffhunk://#diff-b3225420bd8ccd0828bfb2092fa0da2adf6a1a2692f33aebd950eb866a1494f4L2239-R2232) [[3]](diffhunk://#diff-b3225420bd8ccd0828bfb2092fa0da2adf6a1a2692f33aebd950eb866a1494f4L2259-R2258) [[4]](diffhunk://#diff-b3225420bd8ccd0828bfb2092fa0da2adf6a1a2692f33aebd950eb866a1494f4L2320-R2315) [[5]](diffhunk://#diff-b3225420bd8ccd0828bfb2092fa0da2adf6a1a2692f33aebd950eb866a1494f4L2341-R2335) [[6]](diffhunk://#diff-b3225420bd8ccd0828bfb2092fa0da2adf6a1a2692f33aebd950eb866a1494f4L2416-R2400)
* Updated docstrings throughout to clarify input and output shapes for batch support. [[1]](diffhunk://#diff-b3225420bd8ccd0828bfb2092fa0da2adf6a1a2692f33aebd950eb866a1494f4L1510-R1526) [[2]](diffhunk://#diff-b3225420bd8ccd0828bfb2092fa0da2adf6a1a2692f33aebd950eb866a1494f4L2239-R2232) [[3]](diffhunk://#diff-b3225420bd8ccd0828bfb2092fa0da2adf6a1a2692f33aebd950eb866a1494f4L2320-R2315)

**Testing improvements:**

* Added new tests to `tests/pixelfunc/test_ang_vec.py` that verify all major functions—`ang2vec`, `vec2ang`, `pix2vec`, and `get_all_neighbours`—correctly handle and preserve arbitrary batch dimensions, including round-trip and consistency checks.

**Bug fixes and code cleanup:**

* Fixed shape handling in `vec2ang` and `get_all_neighbours` to avoid unnecessary flattening and reshaping, improving clarity and correctness for both scalar and array inputs. [[1]](diffhunk://#diff-b3225420bd8ccd0828bfb2092fa0da2adf6a1a2692f33aebd950eb866a1494f4L1510-R1526) [[2]](diffhunk://#diff-b3225420bd8ccd0828bfb2092fa0da2adf6a1a2692f33aebd950eb866a1494f4L2165) [[3]](diffhunk://#diff-b3225420bd8ccd0828bfb2092fa0da2adf6a1a2692f33aebd950eb866a1494f4L2179-R2199)

### CODE REVIEW

**Comment by ASKabalan** [comment]
## Goal

The main goal is to avoid excessive and unneeded reshaping because in distributed setup .. jax healpy can work out of the box with no extra work on our part .. 

However if you reshape this can cause at worse a communication but most of the time it causes a local copy or a bitcast to reorder the array (so an unnessecary copy)

example 

```python
import jax
jax.config.update('jax_num_cpu_devices', 4)
jax.config.update('jax_platform_name', 'cpu')

import jax.numpy as jnp
import jax_healpy as jhp
from jax import lax
from jax.sharding import Mesh, NamedSharding, PartitionSpec as P
from jax.sharding import AxisType

mesh = jax.make_mesh((2, 2), ("X", "Y"),
                     axis_types=(AxisType.Auto, AxisType.Auto))

sharding = NamedSharding(mesh, P("X", "Y"))
points = jax.random.normal(jax.random.PRNGKey(0), (4, 4 ,4 ,3))
points = lax.with_sharding_constraint(points, sharding)
print(f"Sharding of points is {points.sharding.spec}")


def fn(points):
    theta, phi = jhp.vec2ang(points)
    print(f"Shape and sharding of theta is {theta.shape}, {theta.sharding.spec}")
    print(f"Shape and sharding of phi is {phi.shape}, {phi.sharding.spec}")
    return theta, phi

a = fn(points)


```


With old code this gives


```
Sharding of points is PartitionSpec('X', 'Y')
Shape and sharding of theta is (64,), PartitionSpec('X',)
Shape and sharding of phi is (64,), PartitionSpec('X',)
```

With new code this gives 


```
Sharding of points is PartitionSpec('X', 'Y')
Shape and sharding of theta is (4, 4, 4), PartitionSpec('X', 'Y')
Shape and sharding of phi is (4, 4, 4), PartitionSpec('X', 'Y')
```

**Comment by ASKabalan** [comment]
@pchanial can be reviewed
